File: Day4-PowerOfFour/Solution.cpp
// Runtime: 4 ms
// Memory Usage: 6 MB

class Solution {
public:
    bool isPowerOfFour(int num) {
        if (num<1) return false;
        int i=0,n=num;
        while (n>1){
            n>>=2;
            i+=2;
        }
        return num==(n<<i);
    }
};

File: Day4-PowerOfFour/Solution.py
# Runtime: 28 ms
# Memory Usage: 12.7 MB

class Solution(object):
    def isPowerOfFour(self, num):
        n = 4
        if num==1 : return True
        while n<=num:
            if n==num : return True
            n = n*4
        return False
